### ISSUE TITLE
[Bug] Unity Editor: NotImplementedException on System.Net.ServicePoint.set_ConnectionLeaseTimeout

### ISSUE THREAD

**Issue by najadojo** [AzureAD/microsoft-authentication-library-for-dotnet#2537]
**Logs and Network traces**
N/A

**Which Version of MSAL are you using ?**
MSAL 4.29.0
MSAL 4.10.0 did not exhibit this issue.

**Platform**
net45 in the Unity Editor

**What authentication flow has the issue?**
* Desktop / Mobile
    * [X] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App 
    * [ ] Authorization code
    * [ ] OBO
* Daemon App 
    * [ ] Service to Service calls

Other? - please describe;

**Is this a new or existing app?**
a. The app is in production, and I have upgraded to a new version of MSAL

**Repro**
In a Unity Editor script do roughly the following:

```csharp
PublicClientApplicationBuilder builder = PublicClientApplicationBuilder.Create(clientId);
builder = builder.WithAuthority(AadAuthorityAudience.AzureAdAndPersonalMicrosoftAccount);
builder = builder.WithLogging(MsalLog, LogLevel.Error, true);
var PublicClientApp = builder.Build();

AcquireTokenInteractiveParameterBuilder builder2 = PublicClientApp.AcquireTokenInteractive(scopes);
builder2 = builder2.WithUseEmbeddedWebView(false);
AuthenticationResult authResult = await builder.ExecuteAsync();
```

**Expected behavior**
authResult is returned without exception.

**Actual behavior**
Exception thrown:
```
NotImplementedException: The method or operation is not implemented.
System.Net.ServicePoint.set_ConnectionLeaseTimeout (System.Int32 value) (at <ef151b6abb5d474cb2c1cb8906a8b5a4>:0)
Microsoft.Identity.Client.Platforms.net45.Http.DnsSensitiveClientHandler.AddConnectionLeaseTimeout (System.Uri endpoint) (at D:/a/1/s/src/client/Microsoft.Identity.Client/Platforms/netdesktop/Http/DnsSensitiveClientHandler.cs:52)
Microsoft.Identity.Client.Platforms.net45.Http.DnsSensitiveClientHandler+<SendAsync>d__2.MoveNext () (at D:/a/1/s/src/client/Microsoft.Identity.Client/Platforms/netdesktop/Http/DnsSensitiveClientHandler.cs:28)
```

**Possible Solution**
Don't attempt to set the ConnectionLeaseTimeout on Unity. Also acceptable try catch and log the NotImplementedException during the AddConnectionLeaseTimeout call.

**Additional context/ Logs / Screenshots**

**Comment by bgavrilMS** [comment]
Keeping it open until we release MSAL. @najadojo - we were thinking of releasing next week, is that ok for you? I can send you a package today if you like, but we try not to put preview packages on nuget

**Comment by najadojo** [comment]
@bgavrilMS Thanks for taking my change. I've been on vacation but we also don't need to take an update right away.

**Comment by pmaytak** [comment]
This is included in [MSAL 4.30.0 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.30.0).

cc: @najadojo
